### PR TITLE
Allowing faster graceful shutdown and startup

### PR DESCRIPTION
**Pull request by onukristo**
### Added

* Support for Spring Boot 3.0.

### Changed

- Replaced `@Validated` annotation with custom call to validator.
  `@Validated` annotation based approach made services startup slow.
- Improved the graceful shutdown speed to be less than medium delay interval (by default 5s).
- Changed `MySqlTaskDao` to `JdbcTaskDao`, because some Postgres users got confused/spooked having "mysql" in their stack trace.

### Removed

* Support for Spring Boot 2.5.

## Checklist
- [x] Change meets or does not compromise the [Baseline Security Requirements](https://transferwise.atlassian.net/wiki/spaces/EKB/pages/434929973/Baseline+Security+Requirements)

### CODE REVIEW

File: demoapp/src/main/java/com/transferwise/tasks/demoapp/Application.java [hunk 1]
```diff
@@ -4,11 +4,9 @@
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 import org.springframework.boot.context.ApplicationPidFileWriter;
-import org.springframework.transaction.annotation.EnableTransactionManagement;
 import org.springframework.web.servlet.config.annotation.EnableWebMvc;
 
 @SpringBootApplication
-@EnableTransactionManagement
```
Comment (tw-peeterkarolin) [inline demoapp/src/main/java/com/transferwise/tasks/demoapp/Application.java:11]: This is handled by gaffer starter now right?

File: tw-tasks-kafka-listener-spring-boot-starter/src/main/java/com/transferwise/tasks/ext/kafkalistener/autoconfigure/TwTasksExtKafkaListenerAutoConfiguration.java [hunk 2]
```diff
@@ -51,7 +50,6 @@ public KafkaListenerMetricsTemplate twTasksKafkaListenerMetricsTemplate(IMeterCa
   }
 
   @Bean
-  @Validated
```
Comment (tw-peeterkarolin) [inline tw-tasks-kafka-listener-spring-boot-starter/src/main/java/com/transferwise/tasks/ext/kafkalistener/autoconfigure/TwTasksExtKafkaListenerAutoConfiguration.java:54]: This properties is no longer validated? I can't see the explicit validator classes in this subproject.

File: demoapp/src/main/java/com/transferwise/tasks/demoapp/Application.java (hunk 1 above)
Comment (onukristo) [reply demoapp/src/main/java/com/transferwise/tasks/demoapp/Application.java:11]: Yep,
```
@EnableTransactionManagement
public class GafferJtaConfiguration {
```

File: tw-tasks-kafka-listener-spring-boot-starter/src/main/java/com/transferwise/tasks/ext/kafkalistener/autoconfigure/TwTasksExtKafkaListenerAutoConfiguration.java (hunk 2 above)
Comment (onukristo) [reply tw-tasks-kafka-listener-spring-boot-starter/src/main/java/com/transferwise/tasks/ext/kafkalistener/autoconfigure/TwTasksExtKafkaListenerAutoConfiguration.java:54]: The whole module has been deprecated for a long time and should not be used. However codesearch shows it is still widely used in Wise.
I'll add the programmatic validation here as well.

**Review by tw-peeterkarolin — APPROVED** [review]

**Pull request merged**